src/nvim/strings.c

```diff
@@ -574,6 +574,20 @@ char *concat_str(const char *restrict str1, const char *restrict str2)
   return dest;
 }
 
+/// Concatenate two strings with separator in middle and return the result in allocated memory.
+char *join_str(const char *restrict str1, const char *restrict str2, const char *restrict sep)
+  FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC FUNC_ATTR_NONNULL_ALL
+{
+  size_t l = strlen(str1);
+  size_t s = strlen(sep);
+  char *dest = xmalloc(l + s + strlen(str2) + 1);
+  STRCPY(dest, str1);
+  STRCPY(dest + l, sep);
```

Comment:
since `join_str` isn't joining a list (`ga_concat_strings_sep` does that already), it's probably not that useful. the normal way to do this would be like:

    a2 = concat_str(a, ",")
    a3 = concat_str(a2, b)

Reply:
> (ga_concat_strings_sep does that already),

Great. I was so sure there has to be equivalent of this already in the codebase. But couldn't find it. I'll move it to reuse ga_concat_strings_sep. Or the above double call as you've suggested. I was trying to reduce the number of copies we make I think.